kernel/src/pci.c

```diff
@@ -46,8 +46,10 @@ uint32_t pci_read_config_u32(uint32_t base_address, uint8_t offset) {
     return config_data;
 }
 
-// returns vendor ID, garuanteed to be 0xffff iff device dosen't exist.
-uint16_t get_vendor(uint32_t address) { return pci_read_config_u16(address, VENDOR_ID_OFFSET); }
+// uses vendor ID to check if device exists
+bool device_exists(uint32_t address) {
+    return pci_read_config_u16(address, VENDOR_ID_OFFSET != 0xffff)
```

Comment:
The comparision should be on the function return value and not the constant

Reply:
yes. im dumb ((: